File: CreacionXml/Variables.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using System.Xml;

namespace CreacionXml
{
    public class Variables
    {
        public static string PrettyXML(string xmlString)
        {
            StringWriter sw = new StringWriter();
            XmlTextWriter xw = new XmlTextWriter(sw);
            xw.Formatting = Formatting.Indented;
            xw.Indentation = 4;

            XmlDocument doc = new XmlDocument();
            doc.LoadXml(xmlString);
            doc.Save(xw);

            return sw.ToString();
        }

    }
}

File: Fuente de FE/CrearXML/variables.vb
Imports System.IO
Imports System.Xml

Module variables
    Public Function PrettyXML(XMLString As String) As String
        Dim sw As New StringWriter()
        Dim xw As New XmlTextWriter(sw)
        xw.Formatting = Formatting.Indented
        xw.Indentation = 4
        Dim doc As New XmlDocument
        doc.LoadXml(XMLString)
        doc.Save(xw)
        Return sw.ToString()
    End Function
End Module
